### ISSUE TITLE
Bug: Built-in HTTP Proxy Server Not redirecting from http to https

### ISSUE THREAD

**Issue by aleksannikit** [qdm12/gluetun#1620]
### Is this urgent?

None

### Host OS

Debian 11

### CPU arch

aarch64

### VPN service provider

PureVPN

### What are you using to run the container

Portainer

### What is the version of Gluetun

2023-05-27T18:05:47.959Z (commit 7ba70f1)

### What's the problem 🤔

**Built-in HTTP Proxy Server gives an error 500, instead of redirect when accessing http resources that redirect to https.**

Example:
C:\Users\user>curl -v -x "http://192.168.199.195:8888" http://duckduckgo.com/
*   Trying 192.168.199.195:8888...
* Connected to 192.168.199.195 (192.168.199.195) port 8888 (#0)
> GET http://duckduckgo.com/ HTTP/1.1
> Host: duckduckgo.com
> User-Agent: curl/8.0.1
> Accept: */*
> Proxy-Connection: Keep-Alive
>
< HTTP/1.1 500 Internal Server Error
< Content-Type: text/plain; charset=utf-8
< X-Content-Type-Options: nosniff
< Date: Sun, 28 May 2023 06:31:34 GMT
< Content-Length: 13
<
server error
* Connection #0 to host 192.168.199.195 left intact

**Everything is fine inside the container:**

/ # wget http://duckduckgo.com/
--2023-05-28 09:35:23--  http://duckduckgo.com/
Resolving duckduckgo.com (duckduckgo.com)... 52.142.124.215
Connecting to duckduckgo.com (duckduckgo.com)|52.142.124.215|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://duckduckgo.com/ [following]
--2023-05-28 09:35:23--  https://duckduckgo.com/
Connecting to duckduckgo.com (duckduckgo.com)|52.142.124.215|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6031 (5.9K) [text/html]
Saving to: 'index.html'

index.html                            100%[=======================================================================>]   5.89K  --.-KB/s    in 0s      

2023-05-28 09:35:23 (239 MB/s) - 'index.html' saved [6031/6031]

### Share your logs

```plain text
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2023-05-27T18:05:47.959Z (commit 7ba70f1)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-05-28T09:21:08+03:00 INFO [routing] default route found: interface eth0, gateway 172.20.0.1, assigned IP 172.20.0.2 and family v4
2023-05-28T09:21:08+03:00 INFO [routing] local ethernet link found: eth0
2023-05-28T09:21:08+03:00 INFO [routing] local ipnet found: 172.20.0.0/16
2023-05-28T09:21:08+03:00 INFO [firewall] enabling...
2023-05-28T09:21:08+03:00 INFO [firewall] enabled successfully
2023-05-28T09:21:09+03:00 INFO [storage] merging by most recent 13056 hardcoded servers and 13056 servers read from /gluetun/servers.json
2023-05-28T09:21:09+03:00 INFO Alpine version: 3.18.0
2023-05-28T09:21:09+03:00 INFO OpenVPN 2.5 version: 2.5.8
2023-05-28T09:21:09+03:00 INFO OpenVPN 2.6 version: 2.6.4
2023-05-28T09:21:09+03:00 INFO Unbound version: 1.17.1
2023-05-28T09:21:09+03:00 INFO IPtables version: v1.8.9
2023-05-28T09:21:09+03:00 INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: purevpn
|   |   └── Server selection settings:
|   |       ├── VPN type: openvpn
|   |       ├── Countries: netherlands
|   |       └── OpenVPN server selection settings:
|   |           └── Protocol: UDP
|   └── OpenVPN settings:
|       ├── OpenVPN version: 2.5
|       ├── User: [set]
|       ├── Password: [set]
|       ├── Network interface: tun0
|       ├── Run OpenVPN as: root
|       └── Verbosity level: 1
├── DNS settings:
|   ├── DNS server address to use: 127.0.0.1
|   ├── Keep existing nameserver(s): no
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:127.0.0.1/104
|               ├── ::ffff:10.0.0.0/104
|               ├── ::ffff:169.254.0.0/112
|               ├── ::ffff:172.16.0.0/108
|               └── ::ffff:192.168.0.0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Target address: cloudflare.com:443
|   ├── Duration to wait after success: 5s
|   ├── Read header timeout: 100ms
|   ├── Read timeout: 500ms
|   └── VPN wait durations:
|       ├── Initial duration: 6s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   ├── Enabled: yes
|   ├── Listening address: :8888
|   ├── User: 
|   ├── Password: [not set]
|   ├── Stealth mode: no
|   ├── Log: no
|   ├── Read header timeout: 1s
|   └── Read timeout: 3s
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   ├── Process GID: 1000
|   └── Timezone: Europe/Istanbul
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2023-05-28T09:21:09+03:00 INFO [routing] default route found: interface eth0, gateway 172.20.0.1, assigned IP 172.20.0.2 and family v4
2023-05-28T09:21:09+03:00 INFO [routing] adding route for 0.0.0.0/0
2023-05-28T09:21:09+03:00 INFO [firewall] setting allowed subnets...
2023-05-28T09:21:09+03:00 INFO [routing] default route found: interface eth0, gateway 172.20.0.1, assigned IP 172.20.0.2 and family v4
2023-05-28T09:21:09+03:00 INFO [dns over tls] using plaintext DNS at address 1.1.1.1
2023-05-28T09:21:09+03:00 INFO [http proxy] listening on :8888
2023-05-28T09:21:09+03:00 INFO [http server] http server listening on [::]:8000
2023-05-28T09:21:09+03:00 INFO [healthcheck] listening on 127.0.0.1:9999
2023-05-28T09:21:09+03:00 INFO [firewall] allowing VPN connection...
2023-05-28T09:21:09+03:00 INFO [openvpn] OpenVPN 2.5.8 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-05-28T09:21:09+03:00 INFO [openvpn] library versions: OpenSSL 3.1.0 14 Mar 2023, LZO 2.10
2023-05-28T09:21:09+03:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]172.94.87.6:53
2023-05-28T09:21:09+03:00 INFO [openvpn] UDP link local: (not bound)
2023-05-28T09:21:09+03:00 INFO [openvpn] UDP link remote: [AF_INET]172.94.87.6:53
2023-05-28T09:21:10+03:00 WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1549', remote='link-mtu 1550'
2023-05-28T09:21:10+03:00 WARN [openvpn] 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
2023-05-28T09:21:10+03:00 INFO [openvpn] [Secure-Server] Peer Connection Initiated with [AF_INET]172.94.87.6:53
2023-05-28T09:21:11+03:00 INFO [openvpn] TUN/TAP device tun0 opened
2023-05-28T09:21:11+03:00 INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2023-05-28T09:21:11+03:00 INFO [openvpn] /sbin/ip link set dev tun0 up
2023-05-28T09:21:11+03:00 INFO [openvpn] /sbin/ip addr add dev tun0 10.9.15.41/27
2023-05-28T09:21:11+03:00 INFO [openvpn] UID set to nonrootuser
2023-05-28T09:21:11+03:00 INFO [openvpn] Initialization Sequence Completed
2023-05-28T09:21:11+03:00 INFO [dns over tls] downloading DNS over TLS cryptographic files
2023-05-28T09:21:12+03:00 INFO [healthcheck] healthy!
2023-05-28T09:21:13+03:00 INFO [dns over tls] downloading hostnames and IP block lists
2023-05-28T09:21:17+03:00 INFO [healthcheck] unhealthy: dialing: dial tcp4: lookup cloudflare.com on 127.0.0.1:53: read udp 127.0.0.1:59932->127.0.0.1:53: read: connection refused
2023-05-28T09:21:25+03:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun/wiki/Healthcheck)
2023-05-28T09:21:25+03:00 INFO [vpn] stopping
2023-05-28T09:21:25+03:00 ERROR [vpn] cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/commits": context canceled
2023-05-28T09:21:25+03:00 INFO [vpn] starting
2023-05-28T09:21:25+03:00 INFO [firewall] allowing VPN connection...
2023-05-28T09:21:25+03:00 INFO [openvpn] OpenVPN 2.5.8 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-05-28T09:21:25+03:00 INFO [openvpn] library versions: OpenSSL 3.1.0 14 Mar 2023, LZO 2.10
2023-05-28T09:21:25+03:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]5.254.48.85:53
2023-05-28T09:21:25+03:00 INFO [openvpn] UDP link local: (not bound)
2023-05-28T09:21:25+03:00 INFO [openvpn] UDP link remote: [AF_INET]5.254.48.85:53
2023-05-28T09:21:25+03:00 INFO [dns over tls] init module 0: validator
2023-05-28T09:21:25+03:00 INFO [dns over tls] init module 1: iterator
2023-05-28T09:21:25+03:00 INFO [dns over tls] start of service (unbound 1.17.1).
2023-05-28T09:21:36+03:00 INFO [healthcheck] program has been unhealthy for 11s: restarting VPN (see https://github.com/qdm12/gluetun/wiki/Healthcheck)
2023-05-28T09:21:36+03:00 INFO [vpn] stopping
2023-05-28T09:21:36+03:00 INFO [vpn] starting
2023-05-28T09:21:36+03:00 INFO [firewall] allowing VPN connection...
2023-05-28T09:21:36+03:00 INFO [openvpn] OpenVPN 2.5.8 aarch64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-05-28T09:21:36+03:00 INFO [openvpn] library versions: OpenSSL 3.1.0 14 Mar 2023, LZO 2.10
2023-05-28T09:21:36+03:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]172.94.94.7:53
2023-05-28T09:21:36+03:00 INFO [openvpn] UDP link local: (not bound)
2023-05-28T09:21:36+03:00 INFO [openvpn] UDP link remote: [AF_INET]172.94.94.7:53
2023-05-28T09:21:36+03:00 WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1549', remote='link-mtu 1550'
2023-05-28T09:21:36+03:00 WARN [openvpn] 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
2023-05-28T09:21:36+03:00 INFO [openvpn] [Secure-Server] Peer Connection Initiated with [AF_INET]172.94.94.7:53
2023-05-28T09:21:37+03:00 INFO [openvpn] TUN/TAP device tun0 opened
2023-05-28T09:21:37+03:00 INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2023-05-28T09:21:37+03:00 INFO [openvpn] /sbin/ip link set dev tun0 up
2023-05-28T09:21:37+03:00 INFO [openvpn] /sbin/ip addr add dev tun0 10.60.3.34/27
2023-05-28T09:21:37+03:00 INFO [openvpn] UID set to nonrootuser
2023-05-28T09:21:37+03:00 INFO [openvpn] Initialization Sequence Completed
2023-05-28T09:21:37+03:00 INFO [dns over tls] generate keytag query _ta-4a5c-4f66. NULL IN
2023-05-28T09:21:37+03:00 INFO [dns over tls] generate keytag query _ta-4a5c-4f66. NULL IN
2023-05-28T09:21:38+03:00 INFO [dns over tls] ready
2023-05-28T09:21:38+03:00 INFO [healthcheck] healthy!
2023-05-28T09:21:52+03:00 WARN [http proxy] cannot process request for client 192.168.199.114:54404: Get "https://duckduckgo.com/": net/http: use last response
```

### Share your configuration

```yml
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 8888:8888/tcp # HTTP proxy
    volumes:
      - /home/pi/docker/gluetun:/gluetun
    environment:
      - HTTPPROXY=on
      - VPN_SERVICE_PROVIDER=purevpn
      - OPENVPN_USER=user
      - OPENVPN_PASSWORD=pass
      - SERVER_COUNTRIES=Netherlands
      - TZ=Europe/Istanbul
    restart: unless-stopped
```

**Comment by ronny332** [comment]
Same behavior for me. Requests to :80 throw an HTTP 500 error, instead of forwarding the forwarding message to the browser, which would follow it instead.

```
curl -v -x "10.0.0.17:3128" http://google.com
*   Trying 10.0.0.17:3128...
* Connected to 10.0.0.22 (10.0.0.22) port 13128 (#0)
> GET http://google.com/ HTTP/1.1
> Host: google.com
> User-Agent: curl/8.0.1
> Accept: */*
> Proxy-Connection: Keep-Alive
>
< HTTP/1.1 500 Internal Server Error
< Content-Type: text/plain; charset=utf-8
< X-Content-Type-Options: nosniff
< Date: Sun, 28 May 2023 18:22:20 GMT
< Content-Length: 13
<
server error
```

**Comment by aleksannikit** [comment]
This error occurs since version 3.34, version 3.33 works fine!

**Comment by qdm12** [comment]
Fixed by bfe5e4380f27707a9cbd1af4b6788f004c5727a4 for the latest image, and also released tag v3.34.2 incorporating v3.34.1 + that single fix.

`qmcgaw/gluetun` and `qmcgaw/gluetun:v3.34.2` (aliased as `qmcgaw/gluetun:v3.34` and `qmcgaw/gluetun:v3`) will be built and ready in the next 5-10 minutes.
